infra/experimental/sanitizers/ExecSan/execSan.cpp

```diff
@@ -66,6 +66,10 @@ const std::string kTripWire = "/tmp/tripwire";
 const std::string kInjectionError = "Shell injection";
 // Shell corruption bug detected based on syntax error.
 const std::string kCorruptionError = "Shell corruption";
+// The magic string that we'll use to detect arbitrary file open
+const std::string kFzAbsoluteDirectory = "/fz/";
```

Comment:
Nice! I wonder if we could gneralize this though to be able to catch these without having to rely on a fixed string.

Maybe we could report a bug on any access on a root /dir that doesn't exist?

CC @Alan32Liu @jonathanmetzman

Reply:
Yep, it sounds like a good start for a sanitizer!
One similar idea come across Jonathan and I while we were brainstorming yesterday was to secretly create some files in common dirs (`etc/`) without telling the program under test and check if any of them are opened.
We could also use dictionaries if we want to check a particular file.

How to avoid false positive remains an open questions.

Reply:
> Maybe we could report a bug on any access on a root /dir that doesn't exist?

Just pushed that @oliverchang 

>  secretly create some files in common dirs (`etc/`) without telling the program under test and check if any of them are opened. 

Like if some program lists all the files in a directory and opens them ?
Not sure about the attack scenario behind this...

> How to avoid false positive remains an open questions

I think this is a per fuzz-target config option
My yaml fuzz target will try to open arbitrary files with some include command and it is ok
But I do not think my png or pcap fuzz target should do that...

Reply:
>Maybe we could report a bug on any access on a root /dir that doesn't exist?

Has anyone tested to see if this isn't normal? If / is in a program's PATH (not super common, but a reasonable example IMO), calling `Popen` will always look for files that don't exist in `/` to try to execute them.

Reply:
I don't think there should be a trailing slash.

Reply:
> Has anyone tested to see if this isn't normal? If / is in a program's PATH (not super common, but a reasonable example IMO), calling Popen will always look for files that don't exist in / to try to execute them.

It seems pretty rare that '/' would be in a program's PATH, and even if it is, it'd probably go through `stat` first before we call `open` (if at all -- we'll probably just call execve on it instead). We could also check that it's a dir, and not any random file. 

Either way, we can experiment and see what results we get back. I expect projects will need some config/opt-out mechanism for the things we add (even for the command injection one).